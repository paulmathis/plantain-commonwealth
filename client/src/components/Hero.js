import styled, { css } from 'styled-components/macro';
import { desktop } from '../util/mediaQueries';

export default styled.section`
  background: linear-gradient(rgba(0, 0, 0, 0.45), rgba(0, 0, 0, 0.45)), url(${props => props.src});
  background-size: cover;
  background-position: center;
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: white;
  margin-top: 60px;
  height: ${props => (props.small ? '300px' : '570px')};
  height: ${props => (props.xsmall ? '150px' : '570px')};

  h1 {
    font-size: 4em;
    margin: 0;
  }

  h2 {
    font-weight: 400;
  }

  button {
    margin-top: 42px;
  }

  ${desktop(css`
    height: ${props => props.small && '300px'};
  `)};
`;
