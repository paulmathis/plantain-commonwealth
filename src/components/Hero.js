import styled from 'styled-components/macro';

export default styled.section`
  background: linear-gradient(rgba(0, 0, 0, 0.45), rgba(0, 0, 0, 0.45)), url(${props => props.src});
  background-size: cover;
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: white;
  margin-top: 60px;
  height: ${props => (props.small ? '300px' : '570px')};

  h1 {
    font-size: 4em;
    margin: 0;
  }

  h2 {
    font-weight: 400;
  }

  button {
    margin-top: 42px;
  }
`;
