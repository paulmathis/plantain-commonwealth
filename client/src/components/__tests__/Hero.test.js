import 'jest-styled-components';
import React from 'react';
import { render } from 'react-testing-library';
import Hero from '../Hero';

const src = '/somepic.jpg';

test('<Hero />', () => {
  const { container } = render(<Hero src={src} />);
  expect(container.firstChild).toMatchSnapshot();
});

test('<Hero /> small', () => {
  const { container } = render(<Hero src={src} small />);
  expect(container.firstChild).toMatchSnapshot();
});

test('<Hero /> xsmall', () => {
  const { container } = render(<Hero src={src} xsmall />);
  expect(container.firstChild).toMatchSnapshot();
});
