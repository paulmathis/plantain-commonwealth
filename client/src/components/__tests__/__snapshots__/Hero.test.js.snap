// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> 1`] = `
.c0 {
  background: linear-gradient(rgba(0,0,0,0.45),rgba(0,0,0,0.45)),url(/somepic.jpg);
  background-size: cover;
  background-position: center;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: white;
  margin-top: 60px;
  height: 570px;
  height: 570px;
}

.c0 h1 {
  text-align: center;
  font-size: 4em;
  margin: 0;
}

.c0 h2 {
  font-weight: 400;
}

.c0 button {
  margin-top: 42px;
}

<section
  class="c0"
  src="/somepic.jpg"
/>
`;

exports[`<Hero /> small 1`] = `
.c0 {
  background: linear-gradient(rgba(0,0,0,0.45),rgba(0,0,0,0.45)),url(/somepic.jpg);
  background-size: cover;
  background-position: center;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: white;
  margin-top: 60px;
  height: 300px;
  height: 570px;
}

.c0 h1 {
  text-align: center;
  font-size: 4em;
  margin: 0;
}

.c0 h2 {
  font-weight: 400;
}

.c0 button {
  margin-top: 42px;
}

@media (max-width:992px) {
  .c0 {
    height: 300px;
  }
}

<section
  class="c0"
  src="/somepic.jpg"
/>
`;

exports[`<Hero /> xsmall 1`] = `
.c0 {
  background: linear-gradient(rgba(0,0,0,0.45),rgba(0,0,0,0.45)),url(/somepic.jpg);
  background-size: cover;
  background-position: center;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: white;
  margin-top: 60px;
  height: 570px;
  height: 150px;
}

.c0 h1 {
  text-align: center;
  font-size: 4em;
  margin: 0;
}

.c0 h2 {
  font-weight: 400;
}

.c0 button {
  margin-top: 42px;
}

<section
  class="c0"
  src="/somepic.jpg"
/>
`;
